homeassistant/components/xiaomi_miio/alarm_control_panel.py

```diff
@@ -0,0 +1,162 @@
+"""Support for Xiomi Gateway alarm control panels."""
+
+from functools import partial
+import logging
+
+from miio import DeviceException
+
+from homeassistant.components.alarm_control_panel import (
+    SUPPORT_ALARM_ARM_AWAY,
+    AlarmControlPanel,
+)
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ["python-miio==0.5.0.1"]
+
+XIAOMI_STATE_ARMED_VALUE = "on"
+XIAOMI_STATE_DISARMED_VALUE = "off"
+XIAOMI_STATE_ARMING_VALUE = "oning"
```

Comment:
That is the 'raw' value that the gateway returns if it is turning on the alarm, it will blink its light red during a settable amount of time (like 15 seconds) while playing a sound at the beginning and end. This period allows you to leave the house before the alarm is activated.

'oning' is translated to STATE_ALARM_ARMING of HomeAssistant on line 145-146

Reply:
Ok.